DirectInput/README.md

```diff
@@ -0,0 +1,82 @@
+
+You must have a force-feedback device connected to your system in order to run the application.
+
+When you run the application, it displays a window with a crosshair and a black spot in it. Click anywhere within the window's client area to move the black spot. (Note that moving the device itself does not do anything.) FFConst exerts a constant force on the device from the direction of the spot, in proportion to the distance from the crosshair. You can also hold down the mouse button and move the spot continuously.
+
+This sample program enumerates the input devices and acquires the first force-feedback device that it finds. If none are detected, it displays a message and terminates.
+
+When the user moves the black spot, joySetForcesXY function converts the cursor coordinates to a force direction and magnitude. This data is used to modify the parameters of the constant force effect.
+
+Joystick
+--------
+
+![joystick.png](joystick.png)
+
+The Joystick sample program obtains and displays joystick data.
+
+The application polls the joystick for immediate data in response to a timer set inside the dialog procedure.
+
+Building for Windows XP
+=======================
+
+The DirectInput legacy API is available on Windows XP, so this sample can be built using Visual Studio 2012 with the Windows XP compatible Platform Toolset "v110\_xp" included with _Visual Studio 2012 Update 1_. Remove the link references to `DXGUID.LIB`. You will need to locally define the required GUIDs by using `#define INITGUID` before including `dinput.h` in one of the modules.
+
+Notes
+=====
+
+* These samples use DirectInput8. The DirectInput7 API is not supported for x64 native applications. The Windows 8.0 SDK only includes the DirectInput8 link library (`dinput8.lib`), and DirectInput7 was last supported by the DirectX SDK (August 2007) release.
+
+* The Xbox 360 Common Controller driver exposes a legacy HID device for compatability with older DirectInput only applications. Such a device can be used with the Joystick sample as a result.
+
+* This Xbox One driver for Windows also exposes a legacy HID device that behaves exactly as the Xbox 360 Common Controller driver's HID device support.
+
+* Legacy joysticks using a "gameport" are not supported for Windows Vista, Windows 7, or Windows 8. Only HID-based USB devices are supported.
+
+* The "ActionMapper" functionality was removed from DirectInput as of Windows Vista and is no longer supported.
+
+More Information
+================
+
+[DirectX Tool Kit: Now with GamePads](http://blogs.msdn.com/b/chuckw/archive/2014/09/05/directx-tool-kit-now-with-gamepads.aspx)
+
+[Where is the DirectX SDK?](http://blogs.msdn.com/b/chuckw/archive/2012/03/22/where-is-the-directx-sdk.aspx)
+
+[Where is the DirectX SDK (2013 Edition)?](http://blogs.msdn.com/b/chuckw/archive/2013/07/01/where-is-the-directx-sdk-2013-edition.aspx) 
+
+[Games for Windows and DirectX SDK blog](http://blogs.msdn.com/b/chuckw/)
```

Comment:
Sorry, but I've let these URLs rot :(

SHould be:

https://walbourn.github.io/directx-tool-kit-now-with-gamepads/

https://walbourn.github.io/where-is-the-directx-sdk-2015-edition/

https://walbourn.github.io/